src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.cs

```diff
@@ -51,119 +57,709 @@ internal struct TypeUnion
+                new DateTime(1899, 06, 13)
+            };
+
+            DateTime minDate = new DateTime(100, 01, 01, 23, 59, 59, 999);
+            yield return new object[]
+            {
+                CreateVariant(VT_DATE, new UnionTypes { _date = minDate.ToOADate() }),
+                minDate
+            };
+
+            // VT_BSTR => string.
+            yield return new object[]
+            {
+                CreateVariant(VT_BSTR, new UnionTypes { _bstr = IntPtr.Zero }),
+                null
+            };
+
+            IntPtr emptyString = Marshal.StringToBSTR("");
+            yield return new object[]
+            {
+                CreateVariant(VT_BSTR, new UnionTypes { _bstr = emptyString }),
+                ""
+            };
+
+            IntPtr oneLetterString = Marshal.StringToBSTR("a");
+            yield return new object[]
+            {
+                CreateVariant(VT_BSTR, new UnionTypes { _bstr = oneLetterString }),
+                "a"
+            };
+
+            IntPtr twoLetterString = Marshal.StringToBSTR("ab");
+            yield return new object[]
+            {
+                CreateVariant(VT_BSTR, new UnionTypes { _bstr = twoLetterString }),
+                "ab"
+            };
+
+            IntPtr embeddedNullString = Marshal.StringToBSTR("a\0c");
+            yield return new object[]
+            {
+                CreateVariant(VT_BSTR, new UnionTypes { _bstr = embeddedNullString }),
+                "a\0c"
+            };
+
+            // VT_DISPATCH => object.
+            yield return new object[]
+            {
+                CreateVariant(VT_DISPATCH, new UnionTypes { _dispatch = IntPtr.Zero }),
+                null
+            };
+
+            var obj = new object();
+            if (!PlatformDetection.IsNetCore)
+            {
+                IntPtr dispatch = Marshal.GetIDispatchForObject(obj);
+                yield return new object[]
+                {
+                    CreateVariant(VT_DISPATCH, new UnionTypes { _dispatch = dispatch }),
+                    obj
+                };
+            }
+            else
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => Marshal.GetIDispatchForObject(obj));
+            }
+
+            // VT_ERROR => int.
+            yield return new object[]
+            {
+                CreateVariant(VT_ERROR, new UnionTypes { _error = int.MaxValue }),
+                int.MaxValue
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_ERROR, new UnionTypes { _error = 0 }),
+                0
+            };
+
+            yield return new object[]
+            {
+                CreateVariant(VT_ERROR, new UnionTypes { _error = int.MinValue }),
+                int.MinValue
+            };
+
+            // VT_BOOL => bool.
+            yield return new object[]
+            {
+                CreateVariant(VT_BOOL, new UnionTypes { _i1 = 1 }),
```

Comment:
this is incorrect. For VT_BOOL, VARIANT_TRUE == -1 instead of 1

Reply:
I added this test for that specific reason - this is an unknown value so I was checking to see how the CLR would react to it. Makes sense to keep it IMO